be/src/runtime/buffer_control_block.cpp

```diff
@@ -316,7 +316,7 @@ void PipBufferControlBlock::_update_dependency() {
         (_batch_queue_empty || _buffer_rows < _buffer_limit || _is_cancelled)) {
```

Comment:
if (_result_sink_dependency){
if (_batch_queue_empty || _buffer_rows < _buffer_limit || _is_cancelled)) {
_result_sink_dependency->set_ready();
} else {
_result_sink_dependency->block();
}
}

Reply:
The old code it too complicated,  I think it is just an else logic.